core/lib/merkle_tree/src/pruning.rs

```diff
@@ -9,11 +14,11 @@ use crate::{
 
 /// Handle for a [`MerkleTreePruner`] allowing to abort its operation.
 ///
-/// The pruner is aborted once the handle is dropped.
-#[must_use = "Pruner is aborted once handle is dropped"]
+/// The pruner is aborted once the abort method on handle is called.
 #[derive(Debug)]
 pub struct MerkleTreePrunerHandle {
     aborted_sender: mpsc::Sender<()>,
+    target_retained_version: Arc<RwLock<Option<u64>>>,
```

Comment:
As per access patterns, the lock is never held for a long amount of time and has an only reader, so using `Mutex` would be OK here. Alternatively, it's possible to use `AtomicU64` (target retained version set to 0 effectively means no pruning); would allow getting rid of poison checks.

Reply:
I agree mutex would also fit here, though I don't see it being better. I do see advantages of using AtomicU64 though, I completely haven't though about using it, nice!

Reply:
The only downside is that we can't store None in it, but there is a safe default, i.e. 0